artifactory/utils/container/image.go

```diff
@@ -1,49 +1,52 @@
 package container
 
 import (
+	"net/http"
 	"path"
 	"strings"
 
+	"github.com/jfrog/jfrog-client-go/artifactory"
 	"github.com/jfrog/jfrog-client-go/utils/errorutils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 )
 
 type Image struct {
-	// Image name includes the registry domain, image base name and image tag e.g.: https://my-registry/docker-local/hello-world:latest.
+	// Image name includes the registry domain, image base name and image tag e.g.: my-registry:port/docker-local/hello-world:latest.
 	name string
 }
 
+func NewImage(imageTag string) *Image {
+	return &Image{name: imageTag}
+}
+
 // Get image name
 func (image *Image) Name() string {
 	return image.name
 }
 
-// Get image relative path in Artifactory.
-func (image *Image) GetPath() (string, error) {
+// Get image name from tag by removing the prefixed registry hostname.
+func (image *Image) GetImageLongNameWithTag() (string, error) {
 	if err := image.validateTag(); err != nil {
 		return "", err
 	}
-	indexOfFirstSlash := strings.Index(image.name, "/")
+	indexOfLastSlash := strings.Index(image.name, "/")
 	indexOfLastColon := strings.LastIndex(image.name, ":")
-	if indexOfLastColon < 0 || indexOfLastColon < indexOfFirstSlash {
+	if indexOfLastColon < 0 || indexOfLastColon < indexOfLastSlash {
 		log.Info("The image '" + image.name + "' does not include tag. Using the 'latest' tag.")
-		return path.Join(image.name[indexOfFirstSlash:], "latest"), nil
+		image.name += ":latest"
 	}
-	return path.Join(image.name[indexOfFirstSlash:indexOfLastColon], image.name[indexOfLastColon+1:]), nil
+	return image.name[indexOfLastSlash+1:], nil
 }
 
-// Get image name from tag by removing the prefixed registry hostname.
-func (image *Image) GetImageBaseNameWithTag() (string, error) {
-	if err := image.validateTag(); err != nil {
+// Get image base name by removing the prefixed registry hostname and the tag.
+// e.g.: https://my-registry/docker-local/hello-world:latest. -> docker-local/hello-world
+func (image *Image) GetImageLongName() (string, error) {
+	imageName, err := image.GetImageLongNameWithTag()
+	if err != nil {
 		return "", err
 	}
-	indexOfLastSlash := strings.LastIndex(image.name, "/")
-	indexOfLastColon := strings.LastIndex(image.name, ":")
-	if indexOfLastColon < 0 || indexOfLastColon < indexOfLastSlash {
-		log.Info("The image '" + image.name + "' does not include tag. Using the 'latest' tag.")
-		return image.name[indexOfLastSlash+1:] + ":latest", nil
-	}
-	return image.name[indexOfLastSlash+1:], nil
+	tagIndex := strings.Index(imageName, ":")
```

Comment:
Let's retuen an "unexpected image name" error with the image name, in case the tagIndex is -1.
The same goes for the ```GetImageTag``` and ```GetRegistry``` functions.